packages/frontend/src/hooks/echarts/useEcharts.ts

```diff
@@ -252,6 +253,88 @@ const removeEmptyProperties = <T = Record<any, any>>(obj: T | undefined) => {
         {},
     );
 };
+
+const getMinAndMaxReferenceLines = (
+    leftAxisYId: string | undefined,
+    rightAxisYId: string | undefined,
+    bottomAxisXId: string | undefined,
+    resultsData: ApiQueryResults | undefined,
+    series: Series[] | undefined,
+) => {
+    if (resultsData === undefined || series === undefined) return {};
+
+    const getMinAndMaxValues = (axis: string | undefined): number[] => {
+        if (!axis) return [];
+        return resultsData.rows
+            .map((row) => row[axis]?.value?.raw)
+            .reduce<number[]>(
+                (acc, p) => {
+                    if (isNaN(p)) return acc;
+                    const value = parseInt(p);
+                    const min = acc[0] < value ? acc[0] : value;
+                    const max = acc[1] > value ? acc[1] : value;
+                    return [min, max];
+                },
+                [0, 0],
+            );
+    };
+
+    const getMinAndMaxReferenceLineValues = (
+        axis: string,
+        fieldId: string | undefined,
+    ): number[] => {
+        const values = series.flatMap((serie) => {
+            const serieFieldId =
+                axis === 'yAxis' ? serie.encode.yRef : serie.encode.xRef;
+            if (serieFieldId.field !== fieldId) return [];
+
+            if (!serie.markLine) return [];
+            return serie.markLine?.data.reduce<number[]>((acc, data) => {
```

Comment:
this reduce should make it compatible with multiple reference lines (separate branch)